tests/unit/utils_test.go

```diff
@@ -0,0 +1,215 @@
+	expectedDevice := "/dev/sdb"
+
+	device, err := moduleutils.FindDevName(mockFs, volumeID)
+	if err != nil {
+		t.Fatalf("FindDevName returned an error: %v", err)
+	}
+	if device != expectedDevice {
+		t.Errorf("Expected device %s, got %s", expectedDevice, device)
+	}
+}
+
+// multiple matches - confirm it returned first correct match
+func TestFindDevName_MultipleMatches(t *testing.T) {
+	mockFs := &MockFs{
+		MockDirContents: []fs.DirEntry{
+			&mockDirEntry{name: "scsi-36001405e9f123456789abcdef"},
+			&mockDirEntry{name: "scsi-36001405e9f123456789abcdef-extra"},
+		},
+		MockSymlinkPath: "/dev/sdc",
+	}
+	volumeID := "36001405e9f123456789abcdef-and-more"
+	expectedDevice := "/dev/sdc"
+
+	device, err := moduleutils.FindDevName(mockFs, volumeID)
+	if err != nil {
+		t.Fatalf("FindDevName returned an error: %v", err)
+	}
+	if device != expectedDevice {
+		t.Errorf("Expected device %s, got %s", expectedDevice, device)
+	}
+}
+
+ // no matches - "not found" but fails gracefully
+func TestFindDevName_NoMatches(t *testing.T) {
+	mockFs := &MockFs{
+		MockDirContents: []fs.DirEntry{
+			&mockDirEntry{name: "some-other-disk-id"},
+			&mockDirEntry{name: "another-unrelated-disk"},
+		},
+	}
+	volumeID := "36001405e9f123456789abcdef-and-more"
+
+	device, err := moduleutils.FindDevName(mockFs, volumeID)
+	if err != nil {
+		t.Fatalf("FindDevName returned an error: %v", err)
+	}
+	if device != "" {
+		t.Errorf("Expected empty device string for no matches, got %s", device)
+	}
+}
+
+// dir doesnt exist or unreadable
+func TestFindDevName_ReadDirError(t *testing.T) {
+	mockFs := &MockFs{
+		MockDirError: fmt.Errorf("simulated ReadDir error"),
+	}
+	volumeID := "36001405e9f123456789abcdef-and-more"
+
+	_, err := moduleutils.FindDevName(mockFs, volumeID)
+	if err == nil {
+		t.Fatalf("Expected an error from ReadDir, but got nil")
+	}
+	if err.Error() != "simulated ReadDir error" {
+		t.Errorf("Expected 'simulated ReadDir error', got %v", err)
+	}
+}
+
+// empty directory
+func TestFindDevName_EmptyDirectory(t *testing.T) {
+	mockFs := &MockFs{
+		MockDirContents: []fs.DirEntry{},
+	}
+	volumeID := "36001405e9f123456789abcdef-and-more"
+
+	device, err := moduleutils.FindDevName(mockFs, volumeID)
+	if err != nil {
+		t.Fatalf("FindDevName returned an error: %v", err)
+	}
+	if device != "" {
+		t.Errorf("Expected empty device string for empty directory, got %s", device)
+	}
+}
+
+// broken symlink
+func TestFindDevName_BrokenSymlink(t *testing.T) {
+	mockFs := &MockFs{
+		MockDirContents: []fs.DirEntry{
+			&mockDirEntry{name: "scsi-36001405e9f123456789abcdef"},
+		},
+		MockSymlinkError: fmt.Errorf("simulated EvalSymlinks error"),
+	}
+	volumeID := "36001405e9f123456789abcdef-and-more"
+
+	_, err := moduleutils.FindDevName(mockFs, volumeID)
+	if err == nil {
+		t.Fatalf("Expected an error from EvalSymlinks, but got nil")
+	}
+	if err.Error() != "simulated EvalSymlinks error" {
+		t.Errorf("Expected 'simulated EvalSymlinks error', got %v", err)
+	}
+}
+
```

Comment:
The list of tests is pretty complete thanks.

Remove the blank lines as well here.